### ISSUE TITLE
Vim failed to build on CentOS 7.2.1511

### ISSUE THREAD

**Issue by tinyheero** [Linuxbrew/linuxbrew#847]
# Please follow the general troubleshooting steps first:

- [x ] Ran `brew update` and retried your prior step?
- [x ] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?

## Bug reports:

Vim is failing to build on CentOS 7.2.1511 complaining about a "cannot compute sizeof (off_t)" error.

https://gist.github.com/anonymous/9abf0a550e9aff07a089

I followed the standard instructions on the README.md:

```bash
ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/linuxbrew/go/install)"
sudo yum groupinstall 'Development Tools' && sudo yum install curl git irb m4 python-setuptools ruby texinfo bzip2-devel curl-devel expat-devel ncurses-devel zlib-devel
```

Added the following to my ~/.bashrc

```bash
export PATH="$HOME/.linuxbrew/bin:$PATH"
export MANPATH="$HOME/.linuxbrew/share/man:$MANPATH"
export INFOPATH="$HOME/.linuxbrew/share/info:$INFOPATH"
```

And then ran:

```
brew install --override-system-vi --with-client-server vim
```

**Comment by sjackman** [comment]
I don't know the root cause of your error, but `brew install gcc vim` may resolve this issue.

**Comment by tinyheero** [comment]
Thanks. I was hoping to avoid having to install a linuxbrew specific gcc and use the Centos' default gcc.

I installed gcc:

```
brew install gcc
```

Ran into the error `make[1]: *** [auto/if_perl.c] Error 29`. I then installed installed perl:

```
brew install perl
```

After this it, I was able to install vim.

**Comment by sjackman** [comment]
Great! Glad to hear it. I wonder if `brew install gcc` was necessary or not.

**Comment by tinyheero** [comment]
It was actually. The Centos gcc was 4.8-5-4 and now my linuxbrew gcc is 5.3.0.

**Comment by purwarshubh10** [comment]
Hi,
I haved faced same issue in centos 6.6 while upgrading vim to latest vim version(vim 8).
In my machine it is gcc compiler version problem. My previous gcc version in centos 6.6 is 4.8
I have resolved it by upgrading my gcc to gcc 5.5. After that it has installed.

**Comment by sjackman** [comment]
Glad to hear that you resolved it. Here's instructions to install Linuxbrew on CentOS 6:
https://github.com/Linuxbrew/brew/wiki/CentOS6
